src/binder/binder.cpp

```diff
@@ -172,15 +172,36 @@ std::string Binder::getUniqueExpressionName(const std::string& name) {
     return "_" + std::to_string(lastExpressionId++) + "_" + name;
 }
 
-bool Binder::isReservedPropertyName(const std::string& name) {
+static std::unordered_set<std::string> reservedColumnNames = {
+    InternalKeyword::ID,
+    InternalKeyword::LABEL,
+    InternalKeyword::SRC,
+    InternalKeyword::DST,
+    InternalKeyword::DIRECTION,
+    InternalKeyword::LENGTH,
+    InternalKeyword::NODES,
+    InternalKeyword::RELS,
+    InternalKeyword::PLACE_HOLDER,
+    StringUtils::getUpper(InternalKeyword::ROW_OFFSET),
+    StringUtils::getUpper(InternalKeyword::SRC_OFFSET),
+    StringUtils::getUpper(InternalKeyword::DST_OFFSET),
+    StringUtils::getUpper(rdf::PID),
+    StringUtils::getUpper(rdf::IRI),
+};
+
+bool Binder::reservedInColumnName(const std::string& name) {
     auto normalizedName = StringUtils::getUpper(name);
-    if (normalizedName == InternalKeyword::ID) {
-        return true;
-    }
-    if (normalizedName == StringUtils::getUpper(std::string(rdf::PID))) {
-        return true;
-    }
-    return false;
+    return reservedColumnNames.contains(normalizedName);
+}
+
+static std::unordered_set<std::string> reservedPropertyLookup = {
```

Comment:
What's the difference between `reservedPropertyLookup` and `reservedColumnNames`? Is the latter one used by ddl statements and the former one used by read clauses? I don't think the name is clear here. better rename it or add comments to explain it a bit more.